File: JavaScript/003 - Largest prime factor/solution.js
var prime = require('../shared/prime');

function execute() {
    var rest = 600851475143;
    for (var i = 2; rest > 1; i++) {
        if (rest % i === 0 && prime.isPrime(i)) {
            rest /= i;  
        }
    }
    return i - 1;    
}

module.exports.execute = execute;

File: GoLang/003 - Largest prime factor/solution.go
package project_euler

import "../shared"

func execute003() int {
	rest := 600851475143
	i := 2
	for ; rest > 1; i++ {
		if rest%i == 0 && shared.IsPrime(i) {
			rest /= i
		}
	}
	return i - 1
}
